h/views/activity.py

```diff
@@ -115,3 +115,60 @@ def group_leave(request):
     location = request.route_url('activity.search', _query=new_params)
 
     return httpexceptions.HTTPSeeOther(location=location)
+
+
+@view_config(route_name='activity.group_search',
+             request_method='POST',
+             renderer='h:templates/activity/search.html.jinja2',
+             request_param='toggle_user_facet')
+def toggle_user_facet(request):
+    """
+    Toggle the given user from the search facets.
+
+    If the search is not already faceted by the userid given in the
+    "toggle_user_facet" request param then redirect the browser to the same
+    page but with the a facet for this user added to the search query.
+
+    If the search is already faceted by the userid then redirect the browser
+    to the same page but with this user facet removed from the search query.
+
+    """
+    if not request.feature('search_page'):
+        raise httpexceptions.HTTPNotFound()
+
+    userid = request.params['toggle_user_facet']
+    username = util.user.split_user(userid)['username']
+
+    new_params = request.params.copy()
+
+    del new_params['toggle_user_facet']
+
+    search_words = new_params.get('q', '').split()
+    query_facet_for_user = _query_facet_for_user(username)
+    if _faceted_by_user(username, new_params):
+        search_words.remove(query_facet_for_user)
+    else:
+        search_words.append(query_facet_for_user)
+    new_params['q'] = ' '.join(search_words)
+
+    location = request.route_url('activity.group_search',
+                                 pubid=request.matchdict['pubid'],
+                                 _query=new_params)
+
+    return httpexceptions.HTTPSeeOther(location=location)
+
+
+def _query_facet_for_user(username):
+    """
+    Return a search query facet for the given username.
+
+    Returns query facets suitable for adding in to the search bar input.
+    For example for "seanh" return 'user:"seanh"'.
+
+    """
+    return 'user:"{username}"'.format(username=username)
```

Comment:
This is gone now that we're using the existing `parse()` and `unparse()` instead